hub/api/dataset.py

```diff
@@ -41,7 +74,7 @@ def __getitem__(self, item: Union[slice, str, int]):
             if item not in self.tensors:
```

Comment:
In getitem, are we sure we want to convert integer slices into slice? Might lead to confusions.
For eg ds[0:1][0] == ds[0] makes sense but if we do this, we allow ds[0][0]==ds[0][0][0]==ds[0] which doesn't really make a lot of pythonic sense.
Moreover, ds[0:1]["image"] should return [np.ones()] while ds[0]["image"] should return np.ones(), similar to how you would slice into a list.

Reply:
This makes complete sense, but I have no idea how to reconcile it. We could maybe have a `DatasetItem` class, but that feels wholly unnecessary. Maybe we can get away with a special case of the `slice` and then squeezing the primary axis when you compute a tensor with integer slice.

Good catch, though. How does Hub 1.0 do this?

Reply:
In Hub 1.x we have 3 kinds of "slices":- integer, slices, and lists. 
Depending on the type of slice, we determine the behavior. We can skip lists, for now, they'll be needed when we introduce filtering.

Reply:
i like the idea of normalizing + batchifying the output shapes

Reply:
@McCrearyD that's too far. They expect that if a certain shape goes in, the same shape comes out. We can't violate that. The slicing is a special case because it's a language feature, and we should follow suit.